.github/workflows/labeler.yml

```diff
@@ -38,41 +36,97 @@ jobs:
+      with:
+        github-token: ${{ steps.get_token.outputs.token }}
+        project-id: ${{ steps.add-to-project.outputs.project-id }}
+        item-id: ${{ steps.add-to-project.outputs.item-id }}
+        field-name: 'Iteration'
+        field-iteration: true
+
+    - name: Assign new internal pull requests to project in current iteration
+      if: (steps.checkUserMember.outputs.isTeamMember == 'true' || steps.checkUserMember.outputs.isExcluded == 'true') && github.event_name == 'pull_request'
+      run: |
+        project_id=$(gh api graphql -f query='query($project_number:Int!){ organization(login:"elastic"){ projectV2(number:$project_number) { id } } }' -F project_number=${PROJECT_NUMBER_ID} --jq '.data.organization.projectV2.id' )
+        item_id=$(gh project item-add ${PROJECT_NUMBER_ID} --url ${PULL_REQUEST_URL} --owner elastic --format json --jq .id)
+
+        status_field_json=$(gh api graphql -f query='query($project_id:ID!){ node(id:$project_id){ ... on ProjectV2 { fields(first:20) { nodes { ... on ProjectV2SingleSelectField { id name options { id name } } } } } } }' --jq '.data.node.fields.nodes[]|select(.name=="Status")' -F project_id=${project_id})
+        status_field_id=$( echo "$status_field_json" | jq -r '.id' )
+        status_field_value_id=$( echo "$status_field_json" | jq -r ".options[]|select(.name==\"${STATUS}\")|.id" )
+        gh project item-edit --id "${item_id}" --project-id "${project_id}" --field-id "${status_field_id}" --single-select-option-id "${status_field_value_id}"
+
+        agent_field_json=$(gh api graphql -f query='query($project_id:ID!){ node(id:$project_id){ ... on ProjectV2 { fields(first:20) { nodes { ... on ProjectV2SingleSelectField { id name options { id name } } } } } } }' --jq '.data.node.fields.nodes[]|select(.name=="Agent")' -F project_id=${project_id})
+        agent_field_id=$(echo "$agent_field_json" | jq -r '.id' )
+        agent_field_value_id=$(echo "$agent_field_json" | jq -r ".options[]|select(.name==\"${AGENT}\")|.id" )
+        gh project item-edit --id "${item_id}" --project-id "${project_id}" --field-id "${agent_field_id}" --single-select-option-id "${agent_field_value_id}"
+
+        iteration_field_json=$(gh api graphql -f query='query($project_id:ID!){ node(id:$project_id) { ... on ProjectV2 { fields(first: 20) { nodes { ... on ProjectV2IterationField { id name configuration { iterations { startDate id } } } } } } } }' --jq '.data.node.fields.nodes[] | select( .name == "Iteration" )' -F project_id=${project_id})
+        iteration_field_id=$(echo "$iteration_field_json" | jq -r '.id' )
+        iteration_field_value_id=$(echo "$iteration_field_json" | jq -r 'last(.configuration.iterations[]|select(.startDate<=(now|strftime("%Y-%m-%d"))))|.id')
+        gh project item-edit --id "${item_id}" --project-id "${project_id}" --field-id "${iteration_field_id}" --iteration-id "${iteration_field_value_id}"
```

Comment:
@v1v for now setting the field values does not work with the "project-field-set" action, however here this part is working as expected, removing the `if` condition allows to make it run with the expected result. I'm giving up on this for today, let's continue on this next week when you have time.